### PR TITLE
Make PySOA compatible with newer versions of attrs

### PR DESCRIPTION
**Pull request by miguelsanchez-eb**
Make PySOA compatible with newer versions of attrs

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -32,20 +32,20 @@ matrix:
       script:
         - ./functional.sh verbose
     - stage: build
-      name: "Unit Tests: Python 2.7 with Attrs 18 and 19"
```
Comment (miguelsanchez-eb) [inline .travis.yml:35]: I realized there was a bug here, the comment said we were testing with attrs 19 but we weren't. I've fixed it and included 2 more versions of attrs.

File: .travis.yml [hunk 2]
```diff
@@ -32,20 +32,20 @@ matrix:
...
       script:
-        - tox -e py27-attrs18,py27-flake8,py27-coverage
+        - tox -e py27-attrs18,py27-attrs19,py27-attrs20,py27-attrs21,py27-flake8,py27-coverage
     - stage: build
       name: "Unit Tests: Python 3.5 with and without PyInotify; Pytest 5.1 and 5.3"
       python: "3.5"
       script:
         - tox -e py35,py35-pyinotify,py35-pytest51,py35-pytest53,coverage
     - stage: build
-      name: "Unit Tests: Python 3.7 with Attrs 18 and 19; PyTest 4.6 and 5.3"
```
Comment (miguelsanchez-eb) [inline .travis.yml:45]: I realized there was a bug here, the comment said we were testing with attrs 19 but we weren't. I've fixed it and included 2 more versions of attrs.

File: tox.ini [hunk 3]
```diff
@@ -19,6 +19,8 @@ deps =
     .[testing]
     attrs18: attrs~=18.2
     attrs19: attrs~=19.1
+    attrs20: attrs~=20.3
+    attrs21: attrs~=21.3
```
Comment (eb-octavio) [inline tox.ini:23]: doesn't this allow for version `21.4` which is the undesired one?

Comment (miguelsanchez-eb) [reply tox.ini:23]: You're right, I've fixed it.

**Pull request merged**